generate_reports.py

```diff
@@ -0,0 +1,332 @@
+        get_maximum: Bool:
+            Maximum or minimum value.
+            True for maximum and False for minimum
+            default: True
+        key_index: int:
+            Key of 2nd dimension of array for sorting
+            default: 0
+
+    Returns:
+        value: int:
+            boundary value
+    """
+
+    year_data_list = sorted(year_data_list, reverse=get_maximum,
+                            key=lambda x: x[key_index])
+    temperature = year_data_list[0][0]
+    month = get_month_string_from_number(year_data_list[0][2])
+    year = year_data_list[0][1]
+    return temperature, month, year
+
+
+def get_monthly_graph_data(weather_data, path, month, year):
+    """Loads the month data and returns max temperature,
+    min temperature and len to iterate list
+
+    Arguments:
+        weather_data: dict:
+            Object of class WeatherData
+        path: str:
+            Folder consisting of all data files
+        month: int:
+            Representing month range(1-12)
+        year: int:
+            Representing year
+
+    Returns:
+        maximum_temperature_list: list:
+            List that contains maximum temperature of month
+        minimum_temperature_list: list:
+            List that contains minimum temperature of month
+        len_: int:
+            Returns the length of the list
+    """
+
+    load_month_data(weather_data, path, month, year)
+
+    #  Getting maximum temperature column
+    max_temp_column = get_column_data_from_alias(weather_data,
+                                                 "max_temperature",
+                                                 month, year)
+
+    #  Getting minimum temperature column
+    min_temp_column = get_column_data_from_alias(weather_data,
+                                                 "min_temperature",
+                                                 month, year)
+
+    max_temp_col_len = len(max_temp_column)
+    min_temp_col_len = len(min_temp_column)
+    len_ = max_temp_col_len
+    if max_temp_col_len != min_temp_col_len:
+        if min_temp_col_len < max_temp_col_len:
+            len_ = min_temp_col_len
+            # change color variables
+            print_str = "\033[1;31;40m Note: Unequal data of maximum"
+            print_str += " and minimum temperature"
+            print(print_str)
+    return max_temp_column, min_temp_column, len_
+
+
+def compute_yearly_weather_report(weather_data: WeatherData,
+                                  path, year: int):
+    """Generate yearly weather report
+
+    Arguments:
+        weather_data: dict:
+            Object of class WeatherData
+        path: str:
+            Folder consisting of all data files
+        year: int:
+            Representing year
+    """
+
+    max_temp_list = []
+    min_temp_list = []
+    max_humidity_list = []
+    min_humidity_list = []
+
+    load_year_data(weather_data, path, year)
+
+    # Checking data of all 12 months
+    for i in range(0, 12):
```

Comment:
`month in range(1,13)` can be used , `month` variable  `month = i +1` can be removed